app/controllers/course/duplications_controller.rb

```diff
@@ -17,7 +17,10 @@ def create
 
   def authorize_duplication
     authorize!(:duplicate_from, current_course)
-    authorize!(:duplicate_to_other_instances, current_tenant) if instance_params != current_tenant.id
+    return unless instance_params != current_tenant.id
+
+    authorize!(:duplicate_across_instances, current_tenant)
+    authorize!(:duplicate_across_instances, Instance.find(instance_params))
```

Comment:
This is a good example where it's better to assign `Instance.find(instance_params)` to a variable and name it properly. I don't know what `Instance.find(instance_params)` means here and whyI need to authorize twice unless I dig the code further.

Reply:
After thinking again, yes indeed the naming is quite bad here, I have defined another variable named `destination_tenant` to describe what this is about.

another alternative is actually to rename the instance_params into destination_instance_params, but I am going for the approach above